### PR TITLE
Jss delete image

### LINKED ISSUE #8: Delete Image
#### As a user I want to delete an image.

__Story__:

Our users are awesome at adding images to our web service. Some of our users
add dozens in a single session. Occasionally, however, some users add the wrong
image, and would like a way to remove those images. Please provide our users
with the ability to delete images. Note that at this point we trust our users
not to abuse this feature and delete more images than they should.

__Acceptance criteria__:
- [x] The user is prompted for confirmation prior to deletion.
- [x] Deleted images are removed from the homepage.

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by jschneid**
Closes #8

### CODE REVIEW

File: app/controllers/images_controller.rb [hunk 1]
```diff
@@ -26,10 +26,22 @@ def index
     @images = Image.order(created_at: :desc)
   end
 
+  def destroy
+    image = Image.find_by(id: params[:id])
+
+    if image
+      image.delete
```
Comment (D-D-M) [inline app/controllers/images_controller.rb]: Probably doesn't matter in an app like this, but delete [won't run any callbacks on the model](https://stackoverflow.com/questions/22757450/difference-between-destroy-and-delete) when it runs. So if you have any `before_destroy` or `after_destroy` hooks, they would be ignored in this case. Just a heads up!

File: app/views/images/index.html.erb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 <% content_for :title, 'Image Sharer' %>
-<h1>Image Sharer - Home</h1>
-<p>
-  <%= link_to 'Add a new image', new_image_path %>
-</p>
+
+<% flash.each do |key, value| %>
+  <div class="alert alert-<%= key %>"><%= value %></div>
+<% end %>
```
Comment (D-D-M) [inline app/views/images/index.html.erb]: This is fantastic, but if you extract this out to `application.html.erb`, you can get [flash messages to appear on every page](https://agilewarrior.wordpress.com/2014/04/26/how-to-add-a-flash-message-to-your-rails-page/)!

File: config/routes.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 Rails.application.routes.draw do
   root to: 'images#index'
 
-  resources :images, only: %i[show new create]
+  resources :images, only: %i[show new create destroy]
```
Comment (D-D-M) [inline config/routes.rb]: After years of writing Ruby, I just learned what %i does a month ago – always makes me smile when I see it now :)

File: test/controllers/images_controller_test.rb [hunk 4]
```diff
@@ -100,4 +99,37 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+    # Verify that the success message appears on the page
+    assert_select 'div.alert-success', flash[:success]
+
+    # Verify that the deleted image no longer appears on the page
+    assert_select 'img[src=?]', deleted_image_url, false
+  end
+
+  test 'attempt to delete a nonexistent image' do
+    assert_difference('Image.count', 0) do
+      delete image_url(-1)
```
Comment (D-D-M) [inline test/controllers/images_controller_test.rb]: Beautiful.

File: app/views/images/index.html.erb (hunk 2 above)
Comment (jschneid) [reply app/views/images/index.html.erb]: That makes a lot of sense, thanks for the suggestion! Applied in 075b2859c5732bce3cd892f024e7cab214c2bf4c.

File: app/controllers/images_controller.rb (hunk 1 above)
Comment (jschneid) [reply app/controllers/images_controller.rb]: Ooh, ok. Thanks for calling that out, and for the link where I could read more about the distinction! Changed it to `destroy` in 075b2859c5732bce3cd892f024e7cab214c2bf4c.

File: test/controllers/images_controller_test.rb (hunk 4 above)
Comment (jschneid) [reply test/controllers/images_controller_test.rb]: 😊

**Pull request merged**